trollpy/tests.py

```diff
@@ -0,0 +1,357 @@
+        'sqlalchemy.url': os.environ["DATABASE_URL"]
+    })
+    config.include(".models")
+    settings = config.get_settings()
+    engine = get_engine(settings)
+    Base.metadata.create_all(engine)
+
+    def teardown():
+        testing.tearDown()
+        transaction.abort()
+        Base.metadata.drop_all(engine)
+
+    request.addfinalizer(teardown)
+    return engine
+
+
+@pytest.fixture(scope="function")
+def new_session(sqlengine, request):
+    """Return new session."""
+    session_factory = get_session_factory(sqlengine)
+    session = get_tm_session(session_factory, transaction.manager)
+
+    def teardown():
+        transaction.abort()
+
+    request.addfinalizer(teardown)
+    return session
+
+
+@pytest.fixture
+def dummy_request(new_session, method="GET"):
+    """Instantiate a fake HTTP Request, complete with a database session."""
+    request = testing.DummyRequest()
+    request.method = method
+    request.dbsession = new_session
+    return request
+
+
+@pytest.fixture
+def dummy_post_request(new_session, method="POST"):
+    """Instantiate a fake HTTP Request, complete with a database session."""
+    request = testing.DummyRequest()
+    request.method = method
+    request.dbsession = new_session
+    return request
+
+
+@pytest.fixture
+def testapp():
+    """Create an instance of webtests TestApp for testing routes."""
+    from webtest import TestApp
+    from trollpy import main
+
+    app = main({}, **{'sqlalchemy.url': os.environ["DATABASE_URL"]})
+    testapp = TestApp(app)
+    session_factory = app.registry["dbsession_factory"]
+    engine = session_factory().bind
+    Base.metadata.drop_all(bind=engine)
+    Base.metadata.create_all(bind=engine)
+
+    return testapp
+
+
+@pytest.fixture
+def fill_the_db(testapp):
+    """Fill the database with some model instances."""
+    session_factory = testapp.app.registry["dbsession_factory"]
+    with transaction.manager:
+        dbsession = get_tm_session(session_factory, transaction.manager)
+        user = User(username='amos',
+                    password='password',
+                    first_name='Amos',
+                    last_name='Boldor',
+                    email='amosboldor@gmail.com',
+                    admin=True)
+        kill_score = KillScore(killscore_id='-8',
+                               statement='Prepare to die.')
+        tts = gTTS(text='You live because I allowed it.', lang='en')
+        tts.save('hello.mp3')
+        current_place = os.path.dirname(os.path.abspath(__file__))[:-7]
+        with open(current_place + 'hello.mp3', 'rb') as a:
+            data = a.read()
+        mp3 = Audio(id=1, binary_file=data)
+        dbsession.add(mp3)
+        dbsession.add(kill_score)
+        dbsession.add(user)
+
+# ======== FUNCTIONAL TESTS ===========
+
+
+def test_home_page_pops_up(testapp):
+    """Test that home page get sent correctly."""
+    response = testapp.get('/', status=200)
+    assert response.status_code == 200
```

Comment:
What about testing the home page when your user is authenticated? Shouldn't there be a difference between what data appears on the page depending on whether or not the incoming request has an `authenticated_userid` attribute? What about testing any of the actual data that's supposed to appear on the home page?